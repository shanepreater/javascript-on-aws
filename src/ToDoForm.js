import React from "react"

export class ToDoForm extends React.Component {
    constructor() {
        super()
        this.state = {
            todo: "",
            urgent: false
        }
    }

    handleAddClick = event => {
        const {addItem, nextKey} = this.props
        const {todo, urgent} = this.state

        addItem({
            key: nextKey,
            name: todo,
            urgent,
            done: false
        })
        this.setState({
            todo: "",
            urgent: false
        })
    }

    updateTodo = event => {
        const todo = event.currentTarget.value
        this.setState({
            todo
        })
    }

    toggleUrgent = () => {
        const urgent = this.state.urgent
        this.setState({
            urgent: !urgent
        })
    }

    render  = () => {
        const {todo, urgent} = this.state

        return (
            <form>
              <input name="todo" value={todo} placeholder="Enter task" onChange={this.updateTodo}/>
              <input type="checkbox" name="urgent" check={urgent} onChange={this.toggleUrgent}/> Is it urgent?
              <input type="button" onClick={this.handleAddClick} value="Add" />
            </form>
        )
    }
}