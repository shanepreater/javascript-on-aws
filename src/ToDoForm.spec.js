import React from "react"
import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';

import {ToDoForm} from "./ToDoForm"

configure({
    adapter: new Adapter()
})

describe("Test the ToDo form", () => {
    it("Test the snapshot matches", () => {
        const wrapper = shallow(<ToDoForm />)
        expect(wrapper.snapshot).toMatchSnapshot()
    })
})