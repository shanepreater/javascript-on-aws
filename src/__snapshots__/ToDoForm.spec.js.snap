// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the ToDo form Test the snapshot matches 1`] = `undefined`;
